src/ProductConstructionService/ProductConstructionService.ReproTool/DarcProcessManager.cs

```diff
@@ -0,0 +1,148 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Text.RegularExpressions;
+using Microsoft.DotNet.DarcLib.Helpers;
+using Microsoft.Extensions.Logging;
+using Microsoft.TeamFoundation.Core.WebApi.Types;
+
+namespace ProductConstructionService.ReproTool;
+internal class DarcProcessManager(
+    IProcessManager processManager,
+    ILogger<DarcProcessManager> logger)
+{
+    private string? _darcExePath = null;
+
+    private const string DarcExeName = "darc";
+
+    internal async Task InitializeAsync()
+    {
+        _darcExePath = await Helpers.Which(DarcExeName);
+    }
+
+    internal async Task<ProcessExecutionResult> ExecuteAsync(IEnumerable<string> args)
+    {
+        if (string.IsNullOrEmpty(_darcExePath))
+        {
+            throw new InvalidOperationException($"Call {nameof(InitializeAsync)} before trying to execute a darc command");
+        }
+
+        var a = await processManager.Execute(
+            _darcExePath,
+            [
+                .. args,
+                "--bar-uri", ReproToolConfiguration.PcsLocalUri
+            ]);
+
+        return a;
+    }
+
+    internal async Task<ProcessExecutionResult> DeleteSubscriptionsForChannel(string channelName)
+    {
+        return await ExecuteAsync(["delete-subscriptions", "--channel", channelName, "--quiet"]);
+    }
+
+    internal async Task<IAsyncDisposable> CreateTestChannelAsync(string testChannelName)
+    {
+        try
+        {
+            await ExecuteAsync(["delete-channel", "--name", testChannelName]);
+        }
+        catch (Exception)
+        {
+            // If there are subscriptions associated to the channel then a previous test clean up failed
+            // Run a subscription clean up and try again
+            try
+            {
+                await DeleteSubscriptionsForChannel(testChannelName);
+                await ExecuteAsync(["delete-channel", "--name", testChannelName]);
```

Comment:
Why do we have a method for `DeleteSubscriptionsForChannel` and not for `DeleteChannel` while the former is just used in one place while the latter in quite a few?

Reply:
added